internal/services/gateway/action/user.go

```diff
@@ -810,6 +810,17 @@ func (h *ActionHandler) DeleteUser(ctx context.Context, userRef string) error {
 		return errors.Errorf("user not logged in")
 	}
 
+	userInvitations, _, err := h.configstoreClient.GetUserInvitations(ctx, userRef, 0)
+	if err != nil {
+		return util.NewAPIError(util.KindFromRemoteError(err), err)
+	}
+	for _, invitation := range userInvitations {
+		_, err = h.configstoreClient.DeleteInvitation(ctx, invitation.OrganizationID, userRef)
+		if err != nil {
+			return util.NewAPIError(util.KindFromRemoteError(err), err)
+		}
+	}
+
```

Comment:
This should be done in the configstore that is responsible for maintaining data consistency